services/can_logger/can_logger.py

```diff
@@ -8,51 +7,124 @@
 import sys
 import collections
 import multiprocessing as mp
+import logging
 from datetime import datetime
 from psycopg2 import OperationalError
 from psycopg2.extras import execute_values
 
-
 def write_to_csv(wr_buff, can_bus):
-
     # Write to CSV from write buffer. First item is timestamp, second is
-    # can_id and third is can_data
-
-    with open(f'/data/log/{can_bus}.csv', mode='a') as logfd:
-
-        logcsv = csv.writer(logfd, delimiter=',', quotechar='"',
-                            quoting=csv.QUOTE_MINIMAL)
-
-        logcsv.writerows(wr_buff)
+    # can_bus, third is can_id, and last is can_data
+    # Generate timestamp to rotate the log
+    global log_path
+    #timestamp = rotate_log()
+    while(not log_path):
```

Comment:
Can't this complexity be removed by checking if the log path exists, and if not, creating it before opening the file?

Reply:
Yes. Ideally, right after starting the program.